core/src/test/java/org/apache/calcite/sql/type/SqlTypeFactoryTest.java

```diff
@@ -83,6 +83,37 @@
     assertThat(leastRestrictive.isNullable(), is(true));
   }
 
+  @Test void testLeastRestrictiveForArrays() {
```

Comment:
Maybe we should add tests where leastRestrictive is not computable, to verify it returns the expected result (`null`)

Reply:
thanks, more tests are added